### PR TITLE
xref fixes and improvements

### PR DESCRIPTION
**Pull request by vemv**
Commits:

* Fix the CIDER `xref-find-references` backend to return correct filenames
  * Fixes https://github.com/clojure-emacs/cider/issues/3112
* Fix the `cider-xref-fn-deps` buttons to direct to the right file
  * This fixed the (unrepored) issue of clicking e.g. `clojure/core.clj` and being directed to an unrelate file like `refactor_nrepl/core.clj`.
* Fix the `xref-find-definitions` CIDER backend to return correct filenames
* Improve the presentation of `xref` data
  * Shorten .jar filenames
  * render the cider-xref-fn-deps UI better

It's all tested out locally.

Cheers - V

### CODE REVIEW

File: cider-find.el [hunk 1]
```diff
@@ -275,14 +301,42 @@ thing at point."
...
+                                  (column (nrepl-dict-get info "column"))
+                                  (line (nrepl-dict-get info "line"))
+                                  (friendly-name (cider--xref-extract-friendly-file-name info))
+                                  ;; translate .jar urls and such:
+                                  (buf (cider--find-buffer-for-file filename))
+                                  (bfn (and buf (buffer-file-name buf)))
+                                  (loc (when buf
+                                         ;; favor `xref-make-file-location' when possible, since that way, we can close their buffers.
+                                         (if bfn
+                                             (xref-make-file-location bfn line (or column 0))
```
Comment (vemv) [inline cider-find.el:318]: From what I could see, one can basically make either an xref location based on a filename (which is the best choice, as it doesn't "leak" buffers by opening them) or in a buffer (which is the secondary choice, needed for "special" buffers like those for .jar files).

Checking out xref's source code, it looks as if locations were some sort of implementable/inheritable thing.

So one could create a new xref location type that was lazy, e.g. was a `"jar:file:..."` url that was only open if the user clicked on it.

Could be done some other day. xref.el seemed a bit dense, and I'm not fluent in `cl-lib`.

File: cider-find.el [hunk 2]
```diff
@@ -28,8 +28,10 @@
 
 (require 'cider-client)
 (require 'cider-common)
+(require 'cider-doc)
```
Comment (bbatsov) [inline cider-find.el:31]: Why is this needed here?

File: cider-find.el [hunk 3]
```diff
@@ -241,11 +243,35 @@ thing at point."
   "Return the relevant identifier at point."
   (cider-symbol-at-point 'look-back))
 
+(defun cider--xref-extract-file (dict)
```
Comment (bbatsov) [inline cider-find.el:246]: I wonder if all the `xref` related stuff belong to `cider-find`, which is more focused on definition lookup. Perhaps the `xref-backend` should go somewhere else?

File: cider-find.el [hunk 4]
```diff
@@ -275,14 +301,42 @@ thing at point."
...
+                                         ;; favor `xref-make-file-location' when possible, since that way, we can close their buffers.
+                                         (if bfn
+                                             (xref-make-file-location bfn line (or column 0))
+                                           (xref-make-buffer-location buf (with-current-buffer buf
+                                                                            (save-excursion
+                                                                              (goto-char 0)
+                                                                              (forward-line line)
+                                                                              (move-to-column (or column 0))
+                                                                              (point)))))))
+                                  (should-be-closed? (and
```
Comment (bbatsov) [inline cider-find.el:325]: Names with `?` are not common in Elisp.

File: cider-find.el (hunk 3 above)
Comment (vemv) [reply cider-find.el:246]: I agree that the existing layout was a mixed bag, I'll see if I can move it

**Comment by vemv** [comment]
All feedback satisfied in individual commits.

**Comment by bbatsov** [comment]
Looks great now. Thanks!
